File: c++/1740.cpp
#include <iostream>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  long long N;
  long long s = 1;
  long long result = 0;

  cin >> N;

  while (N != 0) {
    if (N & 1) result += s;
    s *= 3;
    N >>= 1;
  }

  cout << result;

  return 0;
}

File: js/1740.js
const fs = require('fs');

let N = BigInt(fs.readFileSync('example.txt').toString().trim());
let s = 1n;
let result = 0n;

while (N !== 0n) {
  if (N & 1n) result += s;
  s *= 3n;
  N = N >> 1n;
}

console.log(result.toString());
